documentation/projects/proposals/rekognition_data/20240423-implementation_plan_machine_generated_tags_api.md

```diff
@@ -0,0 +1,331 @@
+# 2024-04-23 Implementation Plan: Machine-generated tags in the API
+
+**Author**: @AetherUnbound
+
+<!-- See the implementation plan guide for more information: https://github.com/WordPress/openverse/tree/19791f51c063d0979112f4b9f4eeace04c8cf5ff/docs/projects#implementation-plans-status-in-rfc -->
+<!-- This template is exhaustive and may include sections which aren't relevant to your project. Feel free to remove any sections which would not be useful to have. -->
+
+## Reviewers
+
+<!-- Choose two people at your discretion who make sense to review this based on their existing expertise. Check in to make sure folks aren't currently reviewing more than one other proposal or RFC. -->
+
+- [ ] @krysal
+- [ ] @obulat
+
+## Project links
+
+<!-- Enumerate any references to other documents/pages, including milestones and other plans -->
+
+- [Project Thread](https://github.com/WordPress/openverse/issues/431)
+- [Project Proposal](/projects/proposals/rekognition_data/20240320-project_proposal_rekognition_data.md)
+
+[parse_tags_logic]:
+  https://github.com/WordPress/openverse/blob/df57ab3eb6586502995d6fa70cf417352ec68402/ingestion_server/ingestion_server/elasticsearch_models.py#L196-L206
+
+## Overview
+
+<!-- An overview of the implementation plan, if necessary. Save any specific steps for the section(s) below. -->
+
+The project proposal linked above endeavors to add a new set of
+machine-generated tags to our catalog database as its end goal, however we
+[already have records in our dataset that include machine-generated tags](https://github.com/WordPress/openverse/pull/3948#discussion_r1552301581).
+Nothing currently exists to distinguish these tags from creator-generated ones,
+except for the presence of an `accuracy` value alongside the tag name.
```

Comment:
It is, but it doesn't show up in API results yet!

Reply:
I remember someone said the `clarifai` tags will be deleted, is that the future for these tags? It makes me wonder if the tag provider will be exposed, like we could potentially have machine-generated tags from different providers (AWS, Google, ...) and it will be good to differentiate them. Maybe it's not necessary to include that change in this IP, it could be a thing limited to the catalog, but now that Olga mentions it I'm curious of this detail.

Reply:
> I remember someone said the `clarifai` tags will be deleted, is that the future for these tags?

I don't believe I recall hearing this voiced or spoken anywhere, please share a reference to that conversation if you're able to find it! We _do_ plan to potentially remove existing `clarifai` tags depending on the criteria we determine for handling demographic tags (described in #4040), perhaps that's what you're referencing.

> It makes me wonder if the tag provider will be exposed, like we could potentially have machine-generated tags from different providers (AWS, Google, ...) and it will be good to differentiate them.

That's actually exactly what's being outlined in the [first section of this IP](https://github.com/WordPress/openverse/pull/4189/files#diff-c6ee4fd5837ec2164415c49f11d8b1ec8964329a6b7f6e11c395c08eb10fca0cR77) (see [this section of the preview document](https://docs.openverse.org/_preview/4189/projects/proposals/rekognition_data/20240423-implementation_plan_machine_generated_tags_api.html#expose-provider-in-tags)). 🙂 It's also noted in the ["expected outcomes" for this project](https://docs.openverse.org/_preview/4189/projects/proposals/rekognition_data/20240423-implementation_plan_machine_generated_tags_api.html#expected-outcomes):

> it is expected that we will be able to clearly distinguish which tags returned by the API are machine-generated and **where those tags came from**.

Sara and I also discussed specifically the use of `unstable__provider` (over `provider`) [here](https://github.com/WordPress/openverse/pull/4189#discussion_r1580224389).